hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -16,75 +16,361 @@
-            throw new PreCheckException(UNRESOLVABLE_REQUIRED_SIGNERS);
+    /**
+     * Given a transaction body and schedule store, validate the transaction meets minimum requirements to
+     * be completed.
+     * <p>This method checks that  the Schedule ID is not null, references a schedule in readable state,
+     * and the referenced schedule has a child transaction.
+     * The full set of checks in the {@link #validate(Schedule, Instant, boolean)} method must also
+     * pass.
+     * If all validation checks pass, the schedule metadata is returned.
+     * If any checks fail, then a {@link PreCheckException} is thrown.</p>
+     * @param idToValidate the ID of the schedule to validate
+     * @param scheduleStore data from readable state which contains, at least, a metadata entry for the schedule
+     *     that the current transaction will sign.
+     * @throws PreCheckException if the ScheduleSign transaction provided fails any of the required validation
+     *     checks.
+     */
+    @NonNull
+    protected Schedule preValidate(
+            @NonNull final ReadableScheduleStore scheduleStore,
+            final boolean isLongTermEnabled,
+            @Nullable final ScheduleID idToValidate)
+            throws PreCheckException {
+        if (idToValidate != null) {
+            final Schedule scheduleData = scheduleStore.get(idToValidate);
+            if (scheduleData != null) {
+                if (scheduleData.scheduledTransaction() != null) {
+                    final ResponseCodeEnum validationResult = validate(scheduleData, null, isLongTermEnabled);
+                    if (validationResult == ResponseCodeEnum.OK) {
+                        return scheduleData;
+                    } else {
+                        throw new PreCheckException(validationResult);
+                    }
+                } else {
+                    throw new PreCheckException(ResponseCodeEnum.INVALID_SCHEDULE_ID);
+                }
+            } else {
+                throw new PreCheckException(ResponseCodeEnum.INVALID_SCHEDULE_ID);
+            }
+        } else {
+            throw new PreCheckException(ResponseCodeEnum.INVALID_SCHEDULE_ID);
         }
     }
 
     /**
-     * @param functionality any {@link HederaFunctionality}
-     * @return true if the functionality could possibly be allowed to be scheduled. Some
-     *     functionally may not be in {@link SchedulableTransactionBody} yet but could be in the
-     *     future. The scheduling.whitelist configuration property is separate from this and
-     *     provides the final list of functionality that can be scheduled.
+     * Given a schedule, consensus time, and long term scheduling enabled flag, validate the transaction
+     * meets minimum requirements to be handled.
+     * <p>
+     *     This method checks that the schedule is not null, has not been executed, is not deleted, and is not
+     *     expired as of the current consensus time.
+     * </p>
```

Comment:
(Optional) Potentially easier to read:

```suggestion
     * <p>
     *     This method checks that the schedule is 
     *     <ul>
     *         <li>not null,</li> 
     *         <li>has not been executed,</li> 
     *         <li>is not deleted, and </li>
     *         <li>is not expired, </li>
     *    </ul>
     *   all as of the current consensus time.
     * </p>
```

Reply:
fixed.